### DISCUSSION TITLE
Is it possible to manage custom mimetype to kind mapping?

### DISCUSSION THREAD

**Discussion by LeonSebastianCoimbra** [Studio-42/elFinder#3369] [Q&A] [answered]
Hi all,

Looking at how ElFinder manage the visualization of the mime types i saw that it happens in two points:

- elFinder.js: it declare a map called "kinds" where for every managed mime type it associate a symbolic tag (es. to 'image/png' it is associated 'PNG')
- elfinder.***.js (where instead of *** we have a language): it translate the kind mapped previously into the final language (es. the kind 'PNG' first became 'kindPNG' and then 'PNG Image')

Is it possible to add some custom kinds (and then custom translations) without altering the original files?

For instance, i'd like that the 'image/vnd.dxf' is presented as 'AutoCAD DXF' but in the standard files this mapping is not done... Someone else would like to show it as '3D Drawing' and so on....

If i change the files it works (of course), but i can risk to lose the changes in case of an upgrade to a new version of ElFinder.

So... i was wondering if there is a way to add custom definitions by code, without relaying on the hard-coded files.

Thanks in advance and kind regards.

L.S.C.

**Comment by nao-pon** [comment, marked as answer]
@LeonSebastianCoimbra Yes, you can do that by dynamically changing the members of your elFinder instance in JavaScript.

```javascript
fm.bind('init', function() {
    fm.kinds['image/vnd.dxf'] = 'DXF';
    fm.messages['kindDXF'] = 'AutoCAD DXF';
});
```

By the way, "fm.kinds['image/vnd.dxf'] ='DXF';" It can be obtained from the fm.mimeTypes object, so in the next version we will be able to get it with only "fm.messages ['kindDXF'] ='AutoCAD DXF';".

Online demo: https://jsbin.com/dugoyil/edit?js,output

**Reply by nao-pon** [reply]
I merged #3375. You can try it with a [nightly build](https://github.com/Studio-42/elFinder/archive/2.1.zip).

**Reply by LeonSebastianCoimbra** [reply]
@nao-pon outstanding, thank you very much for your answer, i will try it asap!!
